src/Storages/MergeTree/MergeTreeMarksLoader.cpp

```diff
@@ -167,6 +167,16 @@ MarkCache::MappedPtr MergeTreeMarksLoader::loadMarksImpl()
         }
     }
 
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    std::ranges::for_each(
+        plain_marks,
+        [](auto & plain_mark)
+        {
```

Comment:
Maybe it's a separate issue but Is it not possible to load the "plain marks" with some "readStringLittleEndian" method (l. 155). We then would not need to do the byteswap manually.

Reply:
I agree with the suggestion completely. If you don't mind I'll leave it alone for now, because the reading is done in rather strange ways.